src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -59488,6 +59520,2310 @@ class C {
+                Assert.False(tf1.IsNotNullableIfReferenceType);
+                var attributes = tf1.GetAttributes();
+                Assert.Empty(attributes);
+            }
+        }
+
+        [Fact]
+        public void Constraints_112()
+        {
+            var source1 =
+@"
+#nullable disable
+partial interface I1<TF1>
+{
+}
+
+partial interface I1<TF1>
+{
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Constraints_113()
+        {
+            var source1 =
+@"
+#nullable enable
+partial interface I1<TF1>
+{
+}
+
+partial interface I1<TF1>
+{
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Constraints_114()
+        {
+            var source1 =
+@"
+#nullable disable
+partial interface I1<TF1>
+{
+}
+
+#nullable enable
+partial interface I1<TF1>
+{
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (3,19): error CS0265: Partial declarations of 'I1<TF1>' have inconsistent constraints for type parameter 'TF1'
+                // partial interface I1<TF1>
+                Diagnostic(ErrorCode.ERR_PartialWrongConstraints, "I1").WithArguments("I1<TF1>", "TF1").WithLocation(3, 19)
+                );
+        }
+
+        [Fact]
+        public void Constraints_115()
+        {
+            var source1 =
+@"
+#nullable enable
+partial interface I1<TF1>
+{
+}
+
+#nullable disable
+partial interface I1<TF1>
+{
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (3,19): error CS0265: Partial declarations of 'I1<TF1>' have inconsistent constraints for type parameter 'TF1'
+                // partial interface I1<TF1>
+                Diagnostic(ErrorCode.ERR_PartialWrongConstraints, "I1").WithArguments("I1<TF1>", "TF1").WithLocation(3, 19)
+                );
+        }
+
+        [Fact]
+        public void Constraints_116()
+        {
+            var source1 =
+@"
+#nullable disable
+partial interface I1<TF1>
+{
+}
+
+partial interface I1<TF1> where TF1 : object
+{
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+
+            var i1 = comp1.GlobalNamespace.GetTypeMember("I1");
+            TypeParameterSymbol tf1 = i1.TypeParameters[0];
+            Assert.Null(tf1.IsNotNullableIfReferenceType);
+
+            var source2 =
+@"
+#nullable disable
+partial interface I1<TF1> where TF1 : new()
+{
+}
+
+partial interface I1<TF1> where TF1 : object, new()
+{
+}
+";
+            var comp2 = CreateCompilation(source2);
+
+            comp2.VerifyDiagnostics(
+                // (3,19): error CS0265: Partial declarations of 'I1<TF1>' have inconsistent constraints for type parameter 'TF1'
+                // partial interface I1<TF1> where TF1 : new()
```

Comment:
This is correct, but they differ only in nullability, and so this should be a nullability warning that is suppressed in this disabled context.

Reply:
> This is correct, but they differ only in nullability, and so this should be a nullability warning that is suppressed in this disabled context.  Please add a comment if this is a known issue.

This is existing behavior and is not changed in this PR. The ```object``` constraint is the source of the error. Partial declarations cannot add any constraints.

---
In reply to: [272740346](https://github.com/dotnet/roslyn/pull/34797#discussion_r272740346) [](ancestors = 272740346)